tests/e2e/e2e.go

```diff
@@ -144,6 +158,20 @@ func (te *TestEnvironment) NewEthClient(nodeURI testnet.NodeURI) ethclient.Clien
 	return client
 }
 
+// Create a new private network that is not shared with other tests.
+func (te *TestEnvironment) NewPrivateNetwork() testnet.Network {
+	// Load the shared network to retrieve its path and exec path
+	sharedNetwork, err := local.ReadNetwork(te.NetworkDir)
+	te.require.NoError(err)
+
+	// The private networks dir is under the shared network dir to ensure it
+	// will be included in the artifact uploaded in CI.
+	privateNetworksDir := filepath.Join(sharedNetwork.Dir, PrivateNetworksDirName)
+	te.require.NoError(os.MkdirAll(privateNetworksDir, perms.ReadWriteExecute))
```

Comment:
optional: I prefer to be explicit with error handling 
```suggestion
        err = os.MkdirAll(privateNetworksDir, perms.ReadWriteExecute)
	te.require.NoError(err)
```